File: app/messages/StationMessage.scala
package messages

import messages.StationMessageType.MessageType

case class StationMessage(messageType: MessageType, id: String, updatedTime: Option[Float] = None, updatedName: Option[String] = None, updatedConsole: Option[String] = None, updatedGame: Option[String] = None) {

}

File: app/assets/javascripts/messages/stationMessage.ts
import {StationMessageType} from "./stationMessageType";

interface IStationMessage {

}

export class StationMessage {

    messageType:StationMessageType;

    id: string;

    updatedTime:number;

    updatedName:string;

    updatedConsole:string;

    updatedGame:string;

    constructor(messageType: StationMessageType, id: string, updatedTime: number = 0, updatedName: string = "", updatedConsole: string = "", updatedGame: string = "") {
        this.messageType = messageType;
        this.id = id;
        this.updatedTime = updatedTime;
        this.updatedName = updatedName;
        this.updatedConsole = updatedConsole;
        this.updatedGame = updatedGame;
    }

    static fromJSON(obj:IStationMessage):StationMessage {
        return $.extend({}, obj);
    }
}
